### PR TITLE
[light/jsonrpc] Provide the actual account for `eth_coinbase` RPC and unify error handeling for light and full client

### PR DESCRIPTION
**Pull request by niklasad1**
The previous implementation always provided the `zero address` on
`eth_coinbase` RPCs. 
Now instead, the actual address is returned on
success and an error when no accounts are found!

Note, this behavior is different from the `full client` where we return the ```zero address``` if no accounts are found!

#### Account exists
```bash
$  curl --data '{"method":"eth_coinbase","params":[],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" -X POST localhost:8545
{"jsonrpc":"2.0","result":"0x006e87b3369ab27991cbc8461641db4e7435e139","id":1}
```

#### Account doesn't exist
```bash
curl --data '{"method":"eth_coinbase","params":[],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" -X POST localhost:8545
{"jsonrpc":"2.0","error":{"code":-32023,"message":"No accounts were found","data":"\"\""},"id":1}
```

### CODE REVIEW

**Comment by sorpaas** [comment]
@niklasad1 Do you think it would be a good idea to align full client and light client behavior? Change light client to return zero address when not found, or change full client to return error when not found?

**Comment by niklasad1** [comment]
@sorpaas
 
Yeah, I think we should align the behavior but I don't like the behavior of returning the zero address to indicate errors. So, I would want to return errors in both clients however it might make the `RPC futures` a bit harder to code!

**Comment by sorpaas** [comment]
We've already modified `eth_getLogs` similarly (return errors when parameters are not found). So I think just changing full node's behavior may be fine if we state it clearly in changelogs.

cc @5chdn @tomusdrw

**Comment by tomusdrw** [comment]
Yeah, we should have both methods aligned. I'm ok to include that in the relnotes, "the RPC spec" doesn't really describe error conditions at all (I guess that's the reason we didn't return many errors in the past), so there is much room for improvement here.

I'm ok with returning an error, but we should also document all the error conditions in wiki in some nice readable and consistent format.

File: rpc/src/v1/impls/light/eth.rs [hunk 1]
```diff
@@ -252,7 +252,15 @@ impl<T: LightChainClient + 'static> Eth for EthClient<T> {
 	}
 
 	fn author(&self) -> Result<RpcH160> {
```
[the remark this answers is not in the document]
Comment (tomusdrw) [reply rpc/src/v1/impls/light/eth.rs]: Historical reasons - hash types were not JSON-serializable in the past, I think currently we could get rid of the wrappers and just use the regular types

**Pull request merged**